packages/core/src/tree-view/tree-item.element.ts

```diff
@@ -0,0 +1,229 @@
+  Animatable,
+  animate,
+  AnimationTreeItemExpandName,
+  AriaBooleanAttributeValues,
+  baseStyles,
+  createId,
+  event,
+  EventEmitter,
+  i18n,
+  I18nService,
+  internalProperty,
+  property,
+  querySlot,
+  querySlotAll,
+  reverseAnimation,
+} from '@cds/core/internal';
+import { html, LitElement } from 'lit-element';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { styles } from './tree-item.element.css.js';
+
+/**
+ * Tree view is a hierarchical component that gives users access to a hierarchical set of objects displayed in a the parent-child relationship.
+ *
+ * ```typescript
+ * import '@cds/core/tree-view/register.js';
+ * ```
+ *
+ * ```html
+ *  <cds-tree>
+ *    <cds-tree-item>1</cds-tree-item>
+ *    <cds-tree-item>2</cds-tree-item>
+ *    <cds-tree-item>3</cds-tree-item>
+ *  </cds-tree>
+ * ```
+ *
+ * @beta
+ * @element cds-tree-item
+ * @slot - Content slot for inside the tree item
+ * @slot expand-collapse-icon
+ * @event expandedChange - notify when the user has clicked the expand / collapse button
+ * @event selectedChange - notify when the user has clicked the tree item
+ * @cssprop --animation-duration
+ * @cssprop --animation-easing
+ * @cssprop --background
+ * @cssprop --color
+ * @cssprop --focus-width
+ * @cssprop --font-size
+ * @cssprop --font-weight
+ */
+@animate({
+  expanded: {
+    true: AnimationTreeItemExpandName,
+    false: reverseAnimation(AnimationTreeItemExpandName),
+  },
+})
+export class CdsTreeItem extends LitElement implements Animatable {
+  @i18n() i18n = I18nService.keys.treeview;
+
+  @internalProperty({ type: String, reflect: true, attribute: 'role' })
+  protected role = 'treeitem';
+
+  @property({ type: String })
+  cdsMotion = 'on';
+
+  @event()
+  cdsMotionChange: EventEmitter<string>;
+
+  @internalProperty({ type: String, reflect: true, attribute: 'aria-disabled' })
+  protected ariaDisabled: AriaBooleanAttributeValues | undefined = 'false';
+
+  @internalProperty({ type: String, reflect: true, attribute: 'aria-expanded' })
+  protected ariaExpanded: AriaBooleanAttributeValues | undefined = 'false';
+
+  @internalProperty({ type: String, reflect: true, attribute: 'aria-selected' })
+  protected ariaSelected: AriaBooleanAttributeValues | undefined = 'false';
+
+  @property({ type: Boolean, reflect: true, attribute: 'multi-select' })
+  multiSelect = false;
+
+  @property({ type: Boolean })
+  disabled = false;
+
+  @property({ type: Boolean, reflect: true })
+  expanded = false;
+
+  @property({ type: Boolean, reflect: true })
+  expandable = false;
+
+  @property({ type: Boolean, reflect: true })
+  indeterminate = false;
+
+  @property({ type: Boolean, reflect: true })
+  loading = false;
+
+  @property({ type: Boolean, reflect: true })
+  selected = false;
+
+  // The :scope selector allows us to query the cds-tree-item nodes that are direct children
+  @querySlotAll(':scope > cds-tree-item') private treeItemChildren: NodeListOf<CdsTreeItem>;
```

Comment:
I still want to document my concern that we won't be able to support the case where our users will want to nest cds-tree-item nodes in their framework-specific component wrappers. This may be often attempted when building recursive dynamic trees.
As we talked with Jeeyun, we do have alternative approaches, but I am afraid that not supporting wrapper elements of cds-tree-item will lead to frequent support requests/issues. We need to, at least, document the scenario well.

Reply:
This is a result of the API limitations of custom elements and Angular combined. Custom Elements can only safely access direct children of slots for encapsulation reasons. The Core forms and alerts have similar API limitations.

Almost all major frameworks default or provide container-less components with the exception of Angular. Most outside of Angular will have no issue with wrapping with their own framework components. Looks like there might be [possible workarounds](https://github.com/angular/angular/issues/18877#issuecomment-844354605).

Reply:
Thanks, Cory - that's an awesome workaround, though a bit peculiar. I'll add it to our Support DB scenarios